src/typewrappers.jl

```diff
@@ -97,13 +97,14 @@ mutable struct ArgInfoList <: AbstractVector{SoapySDRArgInfo}
 
     function ArgInfoList(ptr::Ptr{SoapySDRArgInfo}, length::Csize_t)
         this = new(ptr, length)
-        # TODO: see: https://github.com/pothosware/SoapySDR/issues/361
-        #finalizer(SoapySDRArgInfoList_clear, this)
-        this
+        finalizer(this) do this
+            SoapySDRArgInfoList_clear(this)
+        end
+        return this
     end
 end
 
-SoapySDRArgInfoList_clear(s::ArgInfoList) = @GC.preserve s SoapySDRArgInfoList_clear(pointer_from_objref(s), s.length)
+SoapySDRArgInfoList_clear(s::ArgInfoList) = @GC.preserve s SoapySDRArgInfoList_clear(s.ptr, s.length)
```

Comment:
I am never certain on the correct use of `@GC.preserve`. It looks like if we omit this convenience call we can avoid it all together, like: https://github.com/JuliaTelecom/SoapySDR.jl/blame/61e3793c3fa543d7e0139956dc346d248e76fc17/src/typewrappers.jl#L73

Reply:
Right. What I mean is that we could call `SoapySDRArgInfoList_clear(s.ptr, s.length)` directly in the finalizer to avoid the need for a `@GC.preserve` all together.